lib/src/Uncertainty/Algorithm/MetaModel/Kriging/GeneralizedLinearModelAlgorithm.cxx

```diff
@@ -297,89 +320,139 @@ GeneralizedLinearModelAlgorithm::GeneralizedLinearModelAlgorithm (const Numerica
+  , lastReducedLogLikelihood_(SpecFunc::LogMinNumericalScalar)
 {
   // set data & covariance model
   setData(inputSample, outputSample);
-  setCovariance(covarianceModel);
+  setCovarianceModel(covarianceModel);
 
   // Set basis collection
-  if (multivariateBasis.getSize() > 0)  setBasis(multivariateBasis);
+  if (basisCollection.getSize() > 0)  setBasisCollection(basisCollection);
 
   // Set the isoprobabilistic transformation
   setInputTransformation(inputTransformation);
   initializeMethod();
   initializeDefaultOptimizationSolver();
 }
 
-/** set sample  method */
+/* set sample  method */
 void GeneralizedLinearModelAlgorithm::setData(const NumericalSample & inputSample,
     const NumericalSample & outputSample)
 {
   // Check the sample sizes
   if (inputSample.getSize() != outputSample.getSize())
-    throw InvalidArgumentException(HERE) << "In GeneralizedLinearModelAlgorithm::GeneralizedLinearModelAlgorithm, input sample size (" << inputSample.getSize() << ") does not match output sample size (" << outputSample.getSize() << ").";
+    throw InvalidArgumentException(HERE) << "In GeneralizedLinearModelAlgorithm::GeneralizedLinearModelAlgorithm, input sample size=" << inputSample.getSize() << " does not match output sample size=" << outputSample.getSize();
   // Set samples
   inputSample_ = inputSample;
   outputSample_ = outputSample;
 }
 
-/** set covariance method */
-void GeneralizedLinearModelAlgorithm::setCovariance(const CovarianceModel & covarianceModel)
+/* Covariance model accessors */
+void GeneralizedLinearModelAlgorithm::setCovarianceModel(const CovarianceModel & covarianceModel)
 {
+  // Here we can store any modified version of the given covariance model wrt its parameters as it is mainly a parametric template
   const UnsignedInteger inputDimension = inputSample_.getDimension();
   const UnsignedInteger dimension = outputSample_.getDimension();
 
   // Check dimensions of the covariance model
-  // Normal case
-  if ((covarianceModel.getDimension() == dimension) && (inputDimension == covarianceModel.getSpatialDimension()))
+  // There are 4 cases:
+  // 1) Both the spatial dimension and the dimension of the model match the dimensions of the problem, in which case the model is used as-is
+  // 2) The spatial dimension of the model is 1 and different from the spatial dimension of the problem, and the dimension of both the model and the problem are 1. The actual model is a product of the given model.
+  // 3) The spatial dimension of the model and the problem match, but the dimension of the model is 1, different from the dimension of the problem. The actual model is a tensorization of the given model.
+  // 4) The spatial dimension of the model is 1 and different from the spatial dimension of the problem, and the dimension of the model is 1 and different from the dimension of the problem. The actual model is a tensorization of products of the given model.
+  // The other situations are invalid.
+
```

Comment:
I m wondering if we should keep **all** these cases?
If yes we should warn user. Otherwise we keep **only** the first case